util-images/network/netperfbenchmark/pkg/worker/worker.go

```diff
@@ -74,195 +93,222 @@ var (
 	// -e : enhanced reports, gives more metrics for udp.
 	// -s : run in server mode.
 	// -P numOfClients: handle <numOfClients> number of clients before disconnecting.
-	udpClientArguments = []string{"-c", "{destinationIP}", "-u", "-f", "K", "-l", "20", "-b", "1M", "-e", "-i", "1", "-t", "{duration}"}
-	udpServerArguments = []string{"-s", "-f", "K", "-u", "-e", "-i", "{duration}", "-P", "{numOfClients}"}
-	tcpServerArguments = []string{"-s", "-f", "K", "-i", "{duration}", "-P", "{numOfClients}"}
-	tcpClientArguments = []string{"-c", "{destinationIP}", "-f", "K", "-l", "20", "-b", "1M", "-i", "1", "-t", "{duration}"}
+	udpClientArguments = []string{"-c", "{serverPodIP}", "-u", "-f", "K", "-l", "20", "-b", "1M", "-e", "-i", "1", "-t", "{duration}"}
+	udpServerArguments = []string{"-s", "-f", "K", "-u", "-e", "-i", "{duration}", "-P", "{numberOfClients}"}
+	tcpServerArguments = []string{"-s", "-f", "K", "-i", "{duration}", "-P", "{numberOfClients}"}
+	tcpClientArguments = []string{"-c", "{serverPodIP}", "-f", "K", "-l", "20", "-b", "1M", "-i", "1", "-t", "{duration}"}
 	// Siege command args:
 	// -d1 : random delay between 0 to 1 sec.
 	// -t<duration>S : run test for <duration> seconds.
 	// -c1 : one concurrent user.
-	httpClientArguments = []string{"http://" + "{destinationIP}" + ":" + httpPort + "/test", "-d1", "-t" + "{duration}" + "S", "-c1"}
+	httpClientArguments = []string{"http://" + "{serverPodIP}" + ":" + httpPort + "/test", "-d1", "-t" + "{duration}" + "S", "-c1"}
+
+	protocolArgumentMap = map[string]map[string][]string{
+		"client": {
+			ProtocolHTTP: httpClientArguments,
+			ProtocolTCP:  tcpClientArguments,
+			ProtocolUDP:  udpClientArguments,
+		},
+		"server": {
+			ProtocolTCP: tcpServerArguments,
+			ProtocolUDP: udpServerArguments,
+		},
+	}
 )
 
+var gvk = schema.GroupVersionKind{Group: "clusterloader.io", Kind: "NetworkTestRequest", Version: "v1"}
```

Comment:
we shouldn't go directly to v1 - let's start with v1alpha1

Reply:
Modified.